### PR TITLE
202 smtp support

### PR DESCRIPTION
**Pull request by phkroll**

### CODE REVIEW

File: SL/Form.pm [hunk 1]
```diff
@@ -1077,9 +1077,13 @@ sub parse_template {
...
             $noreply              = $myconfig->{email} if !$noreplyemail; # armaghan 2020-03-31 do not use noreply email if not enabled in defaults
+            $mail->{noreplyemail} = $noreplyemail; # Flag for Mailer
 			$mail->{to}           = qq|$self->{email}|;
-			$mail->{from}         = qq|"$myconfig->{name}" <$noreply>|;
-			$mail->{'reply-to'}   = qq|"$myconfig->{name}" <$myconfig->{email}>|;
+			$mail->{from}         = $noreply;
+			$mail->{fromname}     = $myconfig->{name};
+			$mail->{'reply-to'}   = $myconfig->{email};
+            #$mail->{from}         = qq|"$myconfig->{name}" <$noreply>|;
+            #$mail->{'reply-to'}   = qq|"$myconfig->{name}" <$myconfig->{email}>|;
```
Comment (phkroll) [inline SL/Form.pm:1086]: can be removed I guess?

File: SL/Mailer.pm [hunk 2]
```diff
@@ -22,143 +31,122 @@ sub new {
...
-    open(OUT, $out) or return "$out : $!";
-  } else {
-    open(OUT, ">-") or return "STDOUT : $!";
-  }
+  my ($self) = @_;
+
+  #----- CONFIGURATION
+  my $tmpfolder = 'http://domain.net/ledger123/rma/users';
+  my $apikey = 'xkeysib-ad3c502c031275016ba77259cbaad630f58431abfa6eb4938c56e62e7dc64591-UfY14WhVRGN';
+  #----- CONFIG END
```
Comment (phkroll) [inline SL/Mailer.pm:40]: Config will follow?

File: SL/Mailer.pm [hunk 3]
```diff
@@ -22,143 +31,122 @@ sub new {
...
-  
-  print OUT qq|From: $h{from}
-To: $h{to}
-$h{date}$h{'reply-to'}$h{cc}$h{bcc}$h{subject}
-Message-ID: <$msgid>
-$h{notify}X-Mailer: Run my Accounts $self->{version}
-MIME-Version: 1.0
-|;
 
+  my $json = JSON->new->utf8;
```
Comment (phkroll) [inline SL/Mailer.pm:50]: does utf-8 work? Since we usually use ISO-8859-1.

File: SL/Mailer.pm [hunk 4]
```diff
@@ -22,143 +31,122 @@ sub new {
...
-|;
-    if ($self->{message} ne "") {
-      print OUT qq|--${boundary}
-Content-Type: $self->{contenttype}; charset="$self->{charset}"
+  $data->{sender}->{name} = $self->{fromname};
+  $data->{sender}->{email} = $self->{from};
 
-$self->{message}
+  if ($self->{replyto}){
+     $data->{replyTo}->{name} = $self->{replyto};
```
Comment (phkroll) [inline SL/Mailer.pm:58]: shouldn't it be the same as $self->{fromname}; ?

File: SL/Mailer.pm [hunk 5]
```diff
@@ -22,143 +31,122 @@ sub new {
...
+  my $jsonstr = $json->encode($data);
+
+  $commandline = q~
+  curl -s --request POST \
+      --url https://api.sendinblue.com/v3/smtp/email \
+      --header 'accept: application/json' \
+      --header 'api-key:~.$apikey.q~' \
+      --header 'content-type: application/json' \
+      --data '~.$jsonstr.q~' > /dev/null \
+  ~;
```
Comment (phkroll) [inline SL/Mailer.pm:100]: Shall this curl be configurable? Would be cool so we could use any api via curl, couldn't we?

File: SL/Mailer.pm [hunk 6]
```diff
@@ -22,143 +31,122 @@ sub new {
...
+
+  system("$commandline");
 
   return "";
-  
 }
 
 
-sub encode_base64 ($;$) {
+sub send2 {
```
Comment (phkroll) [inline SL/Mailer.pm:108]: obsolete method? Can't find something calling it.

File: SL/Mailer.pm (hunk 5 above)
Comment (ledger123) [reply SL/Mailer.pm:100]: Yes, we can make it. But we need to first check if the field names are also same for that API. We can do it at later stage once we have all this working with sendinblue.

File: SL/Mailer.pm (hunk 6 above)
Comment (ledger123) [reply SL/Mailer.pm:108]: Yes, I kept here for some testing and will remove ultimately.

File: SL/Mailer.pm (hunk 2 above)
Comment (ledger123) [reply SL/Mailer.pm:40]: Yes. exactly. This is for the time being while I am testing everything.

File: SL/Mailer.pm (hunk 3 above)
Comment (ledger123) [reply SL/Mailer.pm:50]: I need to do these and some other test and will confirm.

File: SL/Mailer.pm (hunk 4 above)
Comment (ledger123) [reply SL/Mailer.pm:58]: Yes, you are right. You have sharp eye. I fix it.

**Pull request closed without merging**